frontend/modelarchive/src/main/java/org/pytorch/serve/archive/s3/AWS4SignerBase.java

```diff
@@ -0,0 +1,254 @@
+package org.pytorch.serve.archive.s3;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URL;
+import java.security.MessageDigest;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.SimpleTimeZone;
```

Comment:
Why not use SDK ? https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/  https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk-s3 ?

Reply:
REST API is chosen for S3 SSE-KMS so that it is consistent with existing model registration api. S3 SSE-KMS rest api requests AWS signature 4 for authorization. However, AWS SDK doesn't provide the aws signature 4 calculator. That's why this class is needed.

Reply:
https://docs.amazonaws.cn/en_us/sdk-for-java/v1/developer-guide/examples-crypto-kms.html